### PR TITLE
drop empty arrays

### PR DESCRIPTION
**Pull request by kasperlewau**
Had a use case that obj-clean appeared to take care of brilliantly up until the point where I hit empty arrays. I figured I'd submit a PR and see if I can get away with not having to maintain a fork : ]

Updated the ava assertions & added a space to satisfy the latest xo/ava rules.

### CODE REVIEW

**Comment by SamVerschueren** [comment]
Seems like a legitimate use case. Just wondering if I should make it optional with a flag or something. Have to come up with the use cases where I'm using it for to make sure it's desired behaviour.

**Comment by SamVerschueren** [comment]
Set the `xo` version to `0.16.0`. It dropped support for older Node versions.

**Comment by kasperlewau** [comment]
Have to pin the `ava` version as well. It is failing at 0.12.0.

**Comment by SamVerschueren** [comment]
Could you make this optional? Add a `preserveArrays` option that is `true` by default.

**Comment by kasperlewau** [comment]
Can do!

**Comment by kasperlewau** [comment]
Would you prefer for that to be `clean({}, true|false)` or `clean({}, { preserveArray: false })`?

**Comment by SamVerschueren** [comment]
The options object `clean({}, { preserveArray: false })`

**Comment by kasperlewau** [comment]
👍

File: readme.md [hunk 1]
```diff
@@ -41,6 +44,15 @@ Type: `Object`
 
 The object to clean up.
 
+#### options
+
+##### preserveArrays
+
+Type: `boolean`
```
Comment (SamVerschueren) [inline readme.md]: Add `<br>` at the end of this line

File: readme.md [hunk 2]
```diff
@@ -26,12 +26,15 @@ clean({foo: {bar: 'baz', baz: null, bax: false}});
 
 clean({foo: {bar: 'baz', baz: {}});
 //=> {foo: {bar: 'baz'}}
+
+clean({foo: 'bar', baz: []}, {preserveArrays: false})
```
Comment (SamVerschueren) [inline readme.md]: Add an extra example above this one

```js
clean({foo: 'bar', baz: []});
//=> {foo: 'bar', baz: []}
```

File: index.js [hunk 3]
```diff
@@ -1,13 +1,20 @@
 'use strict';
 var isPlainObject = require('is-plain-object');
-function clean(obj) {
+
+function clean(obj, options) {
+	options = options || {preserveArrays: true};
```
Comment (SamVerschueren) [inline index.js]:
```js
options = Object.assign({
    preserveArrays: true
}, options);
```

**Review by SamVerschueren — CHANGES_REQUESTED** [review]
Some small improvements :). Thanks for working on this!

**Comment by kasperlewau** [comment]
Looks like we broke `0.12` & `0.10` with `Object.assign`.

Not sure why the addition of the `br` in readme didn't get picked up by GitHub as changed. /shrug

**Comment by SamVerschueren** [comment]
Oh you're right. Use [object-assign](https://github.com/sindresorhus/object-assign) for now. And let's ES2015ify after this PR gets merged.

**Comment by kasperlewau** [comment]
I'm cool with that - given that we do an ES2015ification _after_ the merge of this PR and do a major version bump. 

That way I can target the version prior to the major and not suffer from an increased transpilation footprint or having to maintain an `--ignore /node_modules\/(?!dot-prop|obj-clean)/` regex.

Those are only ever issues if the consumer is targeting a browser though. 

I'll push the addition of `object-assign` when I'm off from work :)

**Comment by SamVerschueren** [comment]
Yes, will release first and do major bump after ES2015ify.

**Comment by SamVerschueren** [comment]
Awesome! Thanks for  working on this! Releases as `1.1.0`.

**Comment by kasperlewau** [comment]
Thank you!

**Pull request merged**